impeller/renderer/pipeline.h

```diff
@@ -87,8 +87,73 @@ PipelineFuture<ComputePipelineDescriptor> CreatePipelineFuture(
     const Context& context,
     std::optional<ComputePipelineDescriptor> desc);
 
+/// This is a classed use to check that the input slots of fragment shaders
+/// match the output slots of the vertex shaders.
+/// It's not used at runtime.
+template <typename VertexShaderT, typename FragmentShaderT>
+class ShaderBindingComparator {
```

Comment:
This is a class because it's required for partial specialization.  Also I wish we got better feedback when this failed, but I couldn't figure out how to get multiple static asserts happening in the loop, that's why it returns bool.